chain-api/src/types/TokenBalance.ts

```diff
@@ -150,21 +150,37 @@ export class TokenBalance extends ChainObject {
    */
   @IsOptional()
   @BigNumberArrayProperty()
-  private instanceIds?: Array<BigNumber>;
+  private _instanceIds?: Array<BigNumber>;
 
   @IsOptional()
   @ValidateNested({ each: true })
   @Type(() => TokenHold)
-  private lockedHolds?: Array<TokenHold>;
+  private _lockedHolds?: Array<TokenHold>;
 
   @IsOptional()
   @ValidateNested({ each: true })
   @Type(() => TokenHold)
-  private inUseHolds?: Array<TokenHold>;
+  private _inUseHolds?: Array<TokenHold>;
 
   @BigNumberIsNotNegative()
   @BigNumberProperty()
-  private quantity: BigNumber;
+  private _quantity: BigNumber;
+
+  get instanceIds(){
```

Comment:
This is the attempt to make some of these vars public but readonly

Reply:
I think changing `quantity` to `_quantity` might possibly modify the property name still, and possibly cause issues if it amounts to a schema change and we end up with a mismatch of existing `TokenBalance` objects using `quantity`, and new `TokenBalance` objects using `_quantity`. Would want to test that certainly. 

I wonder if changing `private` to `readonly` could serve your use case instead, while preserving the same write protection intended for these properties in chaincode?

https://www.typescriptlang.org/docs/handbook/2/classes.html#readonly

Reply:
Only problem with just setting to readonly is the properties are actually mutated in the code, which being readonly would prevent. Its like you need a private version to be read/write and a public version that is readonly

Reply:
Maybe I'm misunderstanding your comment about a mismatch of existing TokenBalance object, since the private properties are only accessible inside the class I'm not sure how that would be a problem.

Reply:
I've backed out those changes until a better fix can be implemented

Reply:
> Maybe I'm misunderstanding your comment about a mismatch of existing TokenBalance object, since the private properties are only accessible inside the class I'm not sure how that would be a problem.

I haven't looked yet, but what I mean by this is:

If we looked in the block explorer, found some example `TokenBalance` objects written to chain, what would the names of the properties be and how might this change affect that? 

It would probably be whatever result we see when we serialize an instance of the class. I know properties like `quantity` are part of that serialized data model.